src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1201,19 +1202,9 @@ public SwitchParameter AllMatches
         /// The text encoding to process each file as.
         /// </summary>
         [Parameter]
-        [ValidateNotNullOrEmpty]
-        [ValidateSetAttribute(new string[] {
-            EncodingConversion.Unicode,
-            EncodingConversion.Utf7,
-            EncodingConversion.Utf8,
-            EncodingConversion.Utf32,
-            EncodingConversion.Ascii,
-            EncodingConversion.BigEndianUnicode,
-            EncodingConversion.Default,
-            EncodingConversion.OEM })]
-        public string Encoding { get; set; }
-
-        private System.Text.Encoding _textEncoding;
+        [ArgumentToEncodingTransformationAttribute()]
+        [ArgumentCompleter(typeof(EncodingArgumentCompleter))]
+        public Encoding Encoding = ClrFacade.GetDefaultEncoding();
```

Comment:
You have a few cases where you declare the parameter as a field instead of a property and it doesn't appear that you enforce non-null.  In this case, the null Encoding is passed to new StreamReader which throws an ArgumentNullException.   

In this particular case, the parameter is not marked as mandatory and an exception should not be raised if not provided.

Reply:
I've added `[ValidateNotNullOrEmpty]` to all of these.

Reply:
I've fixed up all the parameters to use [ValidateNotNullOrEmpty]